### PR TITLE
AML-3406 - Build Scripts & ARM template for automatic build

### PR DESCRIPTION
**Pull request by soharadesign**
- Added azure-pipelines.yml definition
- Added ARM template in new format
- Removed a lot of reference-only projects to prevent them from being built automagically.

### CODE REVIEW

File: azure/template.json [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+            "value": ""
+          }
+
+          "customHostName": {
+            "value": ""
+          },
+          "certificateThumbprint": {
+            "value": ""
+          }
+        }
```
Comment (EwanNoble) [inline azure/template.json]: The above few lines look like a dodgy merge

File: azure/template.json [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+                "name": "ConfigurationStorageConnectionString",
+                "value": "[parameters('configurationStorageConnectionString')]"
+              },
+              {
+                "name": "APPINSIGHTS_INSTRUMENTATIONKEY",
+                "value": "[reference('employer-accounts-web-application-insights').outputs.InstrumentationKey.value]"
+              },
+              {
+                "name": "AuthType",
+                "value": "employer"
```
Comment (EwanNoble) [inline azure/template.json]: Does your application use this setting? I think this is specific to das-reservations

File: azure/template.json [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+      "type": "securestring"
+    },
+    "certificateThumbprintsToLoad": {
+      "type": "string"
+    }
+  },
+  "variables": {
+    "deploymentUrlBase":
+      "https://raw.githubusercontent.com/SkillsFundingAgency/das-platform-building-blocks/master/templates/",
+    "resourceNamePrefix":
```
Comment (EwanNoble) [inline azure/template.json]: Formatting

File: azure/template.json (hunk 1 above)
Comment (soharadesign) [reply azure/template.json]: Indeed they are 👍

File: azure/template.json (hunk 2 above)
Comment (soharadesign) [reply azure/template.json]: will remove, was on the base template so I left it on there 👍

File: azure/template.json [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+              }
+            ]
+          },
+          "customHostName": {
+            "value": ""
+          },
+          "certificateThumbprint": {
+            "value": ""
+          }
+
```
Comment (chelnak) [inline azure/template.json]: 🔥

Comment (chelnak) [inline azure/template.json]: 🔥

File: azure/template.json [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+    }, 
+    "storageConnectionString": {
+      "type": "string"
+    },
+    "tokenCertificateThumbprint": {
+      "type": "string"
+    }
+  },
+  "variables": {
+    "deploymentUrlBase": https://raw.githubusercontent.com/SkillsFundingAgency/das-platform-building-blocks/master/templates/",
```
Comment (EwanNoble) [inline azure/template.json]: You've lost a double quote here
```suggestion
    "deploymentUrlBase": "https://raw.githubusercontent.com/SkillsFundingAgency/das-platform-building-blocks/master/templates/",
```

**Comment by EwanNoble** [comment]
I would recommend VS Code with the ARM template extension for doing any work on ARM templates

File: azure/template.json (hunk 5 above)
Comment (soharadesign) [reply azure/template.json]: Pulled in your recommended fix 👍

File: azure/template.json [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+    },
+    "loggingRedisKey": {
+      "type": "string"
+    }, 
+    "storageConnectionString": {
+      "type": "string"
+    },
+    "tokenCertificateThumbprint": {
+      "type": "string"
+    }
```
Comment (EwanNoble) [inline azure/template.json]: Missing a comma, use VS Code 🗡 

```suggestion
    },
```

File: azure/template.json [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+    "logAnalyticsWorkspaceName": {
+      "type": "string"
+    },
+  },
+  "variables": {
+    "deploymentUrlBase": "https://raw.githubusercontent.com/SkillsFundingAgency/das-platform-building-blocks/master/templates/",
+    "resourceNamePrefix": "[toLower(concat('das-', parameters('resourceEnvironmentName'),'-', parameters('serviceName')))]",
+    "employerAccountsWebAppServiceName": "[concat(variables('resourceNamePrefix'), 'web-as')]",
+    "employerAccountsApiAppServiceName": "[concat(variables('resourceNamePrefix'), 'api-as')]",
+    "employerAccountsDatabaseName": "[concat(variables('resourceNamePrefix'), '-db')]"
```
Comment (EwanNoble) [inline azure/template.json]: The above resource names need to be discussed as part of our rollout conversation, we need to consider whether we want to replace the existing resources or not.

File: azure-pipelines.yml [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+  buildPlatform: 'anycpu'
+
+pool:
+  name: 'Continuous Integration 02 - SSD - 160ACU'
+  demands: 
+  - msbuild
+  - visualstudio
+
+steps:
+- task: gittools.gitversion.gitversion-task.GitVersion@3
```
Comment (EwanNoble) [inline azure-pipelines.yml]: If you're interested in using [GitVerison Mainline Development](https://gitversion.readthedocs.io/en/latest/reference/mainline-development/) you can update this task to version 4:
```suggestion
- task: gittools.gitversion.gitversion-task.GitVersion@4
```
then add this file to the root of the repository:
https://github.com/SkillsFundingAgency/das-apply-service/blob/master/GitVersion.yml

Up to you

File: azure-pipelines.yml [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+  buildPlatform: 'anycpu'
+
+pool:
+  name: 'Continuous Integration 02 - SSD - 160ACU'
+  demands: 
+  - msbuild
+  - visualstudio
+
+steps:
+- script: src/RunBuild.bat "Release"
```
Comment (EwanNoble) [inline azure-pipelines.yml]: If we have removed cloud services then we are absolutely not going back to using this script

Comment (EwanNoble) [inline azure-pipelines.yml]: If we having removed the cloud services then we are absolutely not going back to using this build script

Comment (EwanNoble) [reply azure-pipelines.yml]: If you are having trouble with the build process please get us involved

**Pull request merged**